api/domain/ordersDomain.js

```diff
@@ -69,34 +70,37 @@ exports.getById = async (orderId) => {
     }
 }
 
-exports.addProducts = async (order, products) => {
+exports.addProducts = async (order, productsArray) => {
     try {
-        for (let n = 0; n < products.length; n++) {
-            const productsInOrder = await order.getProducts({ where: { id: products[n].id } })
+        for (let n = 0; n < productsArray.length; n++) {
+            const productsInOrder = await order.getProducts({ where: { id: productsArray[n].id } })
 
             if (productsInOrder.length !== 0) continue
-
-            await OrderHelpers.addQuantityAndTotalPrice(order, products[n].id, products[n].quantity)
+            const product = await productRepo.getById(productsArray[n].id)
```

Comment:
can you try using the product from `productsInOrder` instead of getting it from repo?

Reply:
nah, because productInOrder has only priceId, not the prices itself

Reply:
and this is regarding all your comments here about this case